addmeta/addmeta.py

```diff
@@ -95,6 +95,10 @@ def add_meta(ncfile, metadict, template_vars, verbose=False):
 
     # Set global meta data
     if "global" in metadict:
+        if sort_attrs:
+            # Add all attrs to the metadict, sort it and remove attrs from rootgrp
+            metadict = sort_metadict_remove_attrs(metadict, rootgrp)
+
```

Comment:
```suggestion
            # Remove all global attributes, update that dict with new or updated
            # versions, and sort combined dict
            metadict = order_dict(delete_global_metadata(rootgrp).update(metadict))

```
Using `update` on the returned existing globals takes care of duplicates: they're automatically overwritten by new versions in `metadict`.

The `order_dict` function is mostly for neatness, but it does mean you can write a test for it, which is nice.